assets/src/block-templates/register.js

```diff
@@ -34,7 +40,29 @@ export const registerBlockTemplates = blockTemplates => {
 
     registerBlockType(metadata, {
       edit: edit(template, templateLock),
-      save,
+      save: props => {
+        if(props.innerBlocks.length) {
+          const className = `${props.innerBlocks[0].attributes.className || ''}`;
+          let patternClassName = '';
+
+          switch (metadata.type) {
+          case 'planet4':
+            patternClassName = 'planet4-pattern';
+            break;
+          case 'pageHeaders':
+            patternClassName = 'page-headers-pattern';
+            break;
+          case 'layouts':
+            patternClassName = 'planet4-layout-pattern';
+            break;
+          }
+
+          if(!className.includes(patternClassName)) {
+            props.innerBlocks[0].attributes.className = `${className} ${patternClassName}`;
```

Comment:
a little trim on this string to avoid spaces ?

Reply:
I don't see necessary in this case. Why? Since the `${className}` is the default class (as `block`) and you we are re-assigning that. In this case the new `className` would be `block planet4-layout-pattern` or  would be `block planet4-pattern`.